src/release-specification.ts

```diff
@@ -301,6 +299,58 @@ export async function validateReleaseSpecification(
           });
         }
       }
+
+      // Check to compel users to release new versions of dependencies alongside their dependents
+      if (
+        pkg &&
+        versionSpecifierOrDirective &&
+        (hasProperty(IncrementableVersionParts, versionSpecifierOrDirective) ||
+          isValidSemver(versionSpecifierOrDirective))
+      ) {
+        const missingDependencies = Object.keys({
+          ...(pkg.unvalidatedManifest.dependencies || {}),
+          ...(pkg.unvalidatedManifest.peerDependencies || {}),
+        }).filter((dependency) => {
+          const dependencyVersionSpecifierOrDirective =
+            unvalidatedReleaseSpecification.packages[dependency];
+
+          return (
```

Comment:
Note that we should only show this warning if the dependency has unreleased changes. We'll need an additional filter step.

At the moment it seems to expect _all_ dependencies from being listed, even those not in the monorepo!

Reply:
Right, that's a good point. We can use `project.workspacePackages` to look up the package object for a `dependency`, and once we have it we can access its `.hasChangesSinceLatestRelease` property to know whether it's changed.